### PR TITLE
feat: remove next_token from ListResult (#35)

### LINKED ISSUE #35: Remove next_token from ListResult
ListResult has next_token: Option<String> (Token passed to the API for the next page of list results) 
ObjectStore has nowhere a method that accepts such token, thus no idea why this is part of the API.

fn list: List all the objects with the given prefix

fn list_with_delimiter: List objects with the given prefix and an implementation specific delimiter. 
-> Should all such objects be returned? Or only some? (And how many? Leave it to underlying impl?)

### PR DESCRIPTION
**Pull request by tustvold**
This dates from a previous incarnation of the API where pagination was exposed to the user, it can be removed

Closes #35

### CODE REVIEW

File: src/azure.rs [hunk 1]
```diff
@@ -287,95 +292,48 @@ impl ObjectStore for MicrosoftAzure {
     }
 
     async fn list(&self, prefix: Option<&Path>) -> Result<BoxStream<'_, Result<ObjectMeta>>> {
-        #[derive(Clone)]
```
Comment (tustvold) [inline src/azure.rs:290]: This is a drive-by-fix, the listing logic is extracted into a `list_impl` so that it can be used by `list_with_delimiter`

File: src/azure.rs [hunk 2]
```diff
@@ -447,11 +460,13 @@ fn convert_object_meta(blob: Blob) -> Result<ObjectMeta> {
         .try_into()
         .expect("unsupported size on this platform");
 
-    Ok(ObjectMeta {
+    // This is needed to filter out gen2 directories
+    // https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-known-issues#blob-storage-apis
+    Ok((size > 0).then(|| ObjectMeta {
```
Comment (tustvold) [inline src/azure.rs:465]: Another drive-by-fix, this logic is moved to convert_object_meta so that it is applied consistently to all requests

**Review by crepererum — APPROVED** [review]
